src/util/RouterUtil.js

```diff
@@ -133,17 +133,12 @@ function (Okta, OAuth2Util, Util, Enums, BrowserFeatures, Errors, ErrorCodes) {
 
       var redirectFn = router.settings.get('redirectUtilFn');
 
-      if (res.type === Enums.SESSION_STEP_UP) {
-        var targetUrl = res._links && res._links.next && res._links.next.href;
-        successData.stepUp = {
-          url: targetUrl,
-          finish: function () {
-            redirectFn(targetUrl);
-          }
+      var nextUrl = res._links && res._links.next && res._links.next.href;
```

Comment:
Hmm, it's not actually going to be a "next" link but rather an "original" link:

```json
"_links": {
        "original": {
            "href": "https://your-org.okta.com/login/idx/redirect?stateToken=01OCl7uyAUC4CUqHsObI9bvFiq01cRFgbnpJQ1bz82",
            "hints": {
                "allow": [
                   "GET"
                ]
            }
        }
    }
```

Reply:
Actually -- we need to debate this.

@tomabbott-okta @wdawson-okta @robertdamphousse-okta @karlmcguinness-okta 

We have been moving away from the "next" link pattern and moving towards more explicit link names which are the actual operation names. So instead of:
```json
"_links":{
      "next":{
         "name":"original",
         "href":"https://atko.oktapreview.com/login/step-up/redirect?stateToken=00quAZYqYjXg9DZhS5UzE1wrJuQ6KKb_kzOeH7OGB5",
         "hints":{
            "allow":[
               "GET"
            ]
         }
      }
   }
```
you'd have:
```json
"_links": {
        "original": {
            "href": "https://your-org.okta.com/login/idx/redirect?stateToken=01OCl7uyAUC4CUqHsObI9bvFiq01cRFgbnpJQ1bz82",
            "hints": {
                "allow": [
                   "GET"
                ]
            }
        }
    }
```
This is how MFA_REQUIRED state is [already modeled in the Auth API.](https://developer.okta.com/docs/api/resources/authn#response-example-for-primary-authentication-with-public-application-factor-challenge). In general we've basically had explicit link operation names in any embedded structures, and then use the `next` pattern in the top-level links.

IMO we need to move away from the `next` pattern altogether and just always offer explicit operation names as the link names.

Reply:
Agree there might be multiple paths to a SUCCESS. That, to me, is the root of why we need to figure out a different pattern than "next".

Reply:
So while we're here then, could we make the widget handle both `next` and `original` the same way (which is: ignore any `sessionToken` and just redirect to the link)? That should keep it backward compatible with the [AuthN Session Step-Up](https://developer.okta.com/docs/api/resources/authn#sp-initiated-step-up-authentication) functionality, which I'm still hoping we can deprecate but for now needs to keep working.

Reply:
Good for me 👍